python/tvm/relay/frontend/pytorch.py

```diff
@@ -2689,7 +2689,16 @@ def rsub(self, inputs, input_types):
     def embedding(self, inputs, input_types):
         weight = inputs[0]
         indices = inputs[1]
-        return _op.embedding(weight, indices.astype("int32"), axis=0)
+        # Check the type of indices, if it is not int32, cast it to int32
+        if _infer_type(indices).checked_type.dtype != "int32" and _infer_type(indices).checked_type.dtype != "int64":
```

Comment:
There is no need to infer type twice. Just a minor suggestion:
```py
 # Check the type of indices, if it is not int32 or int64, cast it to int32
indicies_dtype = _infer_type(indices).checked_type.dtype
if indicies_dtype != "int32" and indicies_dtype != "int64":
    ...
```

And potentially, we can add warning message for users to know that we did casting to the supported dtype :))